api/src/services/programsService.ts

```diff
@@ -0,0 +1,214 @@
+import db from './db';
+import {
+  Program,
+  ProgramActivity,
+  ProgramWithActivities,
+  CurriculumActivity,
+  ActivityType,
+} from '../models';
+import { DateTime, Duration } from 'luxon';
+
+/**
+ * Returns all programs in the database.
+ *
+ * @returns {Program[]} - all programs, or null if not found
+ */
```

Comment:
I'd like us to think about these comment choices before we commit this pull request. As this is a different pattern than our previous files, is this something that we are committing to as a team moving forward on all service files? To me these comments are a bit redundant as I can gather all this information from the function name and I can clearly see what is being returned. If this is a documentation choice to help future developers than we should be considering adding these guidelines to the contributing guide. My personal recommendation is to remove all comments unless it provides value the code can not provide itself. Happy to agree with the team on the decision moving forward.

Reply:
I agree that it's different, but it's in line with standards we tried to enforce with Cohort 3. I take full responsibility for asking the team to contribute the function header comments and enforcing their use and completion.

I hear you with regard to sane naming of functions and variables being an adequate source of documentation, and I hear you with regard to skepticism regarding both the value-add for commenting as well as modifying the contribution guidelines. Thank you for sharing your feedback.

I'd like to share where my head is at with regard to JSDoc/TSDoc, commenting in general, and the need for commenting in our project vs. the reality that mentees will see in their jobs:

### Where we agree
- I think we absolutely should enforce style guidelines (and be explicit about those guidelines) with regard to the naming of functions and variables. I agree this is a core competency for any programmer, and this should be a bare minimum requirement for documentation.
- I agree with you in past, private conversations that we should also enforce guidelines that keep functions small, making it easy to read through their logic. This is also a bare minimum for documentation.

### My experience with jobs
- I've had jobs where the core product(s) were well-documented, and I've had jobs where there was no documentation. Starting a job having had experience with documentation both made me a better candidate for the position as well as helped me to produce better code.

### Commenting guidelines is an opinionated choice
- We haven't been doing it thus far for any features that ended up shipping.
- We don't currently have a commenting standard we adhere to either.
- There's nothing in the contribution guidelines that says whether we should or should not be commenting code we contribute.

### But we make other opinionated choices in this repo
- There's no guarantee mentees will be working on projects that have unit tests at their jobs.
- There's no guarantee mentees will be using git, or will be using the feature branch workflow vs. centralized, gitflow, or forking workflow.
- There's no guarantee mentees will be using React, Docker, or other technology choices we've made.

### The purpose of Rhizone
- Having the mentees work on Rhizone allows them not only to speak to the experience of working on a team, collaborating, compromising, and creating, it allows them to point to code they've worked on that they can be proud of not only because it's good code but also because it's in a shipping product.
- Enforcing standards with documentation will only increase the pride they have in the code they produce.
- Eventually, if we modify the contributing guidelines (which tbh need updating anyway) to include commenting and all new features have comments and we back-fill the old code with comments, code smell will reduce, onboarding will be easier, and the code overall will look more like other open-source projects.
- Finally, if we have full code coverage for comments, then we can install documentation packages and generate documentation like other best-in-class open-source projects.

### In conclusion
- I think going forward, we should require documentation for contributed functions because it's good for mentees and it's good for Rhizone
- I think that while documentation standards are an opinionated choice, it's an opinion we should support because good documentation is sign of a developer that is team-focused, which is the sort of developer that we want our mentees to grow into.
- I think the peculiarity of there being some things that are documented and some that are not is a temporary issue that will decrease over time, especially if we enforce a documentation standard going forward.
- I'm open to persuasive arguments about this as well as compromise regarding the scope of enforcing a standard like this, but I have pride when I look at this file with its function header comments and I'm really happy with the work the mentees have done.
- We'll fix the inline comments that you've pointed out.